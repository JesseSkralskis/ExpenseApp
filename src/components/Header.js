import React from 'react';
import {NavLink,} from 'react-router-dom'






const Header = () => (
  //navlink allows gives us props that help us style easily the
  //nav we are on using activeclassname and the styling is-active
  <header>
    <h1>Expensify</h1>
    <NavLink exact={true} activeClassName="is-active" to="/">
      Home
    </NavLink>
    <NavLink activeClassName="is-active" to="/create">
      create
    </NavLink>
     
    
  </header>
);

export default Header;