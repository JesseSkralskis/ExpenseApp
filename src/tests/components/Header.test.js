
import {shallow} from 'enzyme'
import React from 'react';
import toJSON from 'enzyme-to-json'
import Header from '../../components/Header';


//need to be able to virtually render our component
//shallow rendering is only concerned with the current component not the children props etc

test('should render Header corectly', () => {
    const wrapper = shallow(<Header/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
    //find function allows us to select element in all ways
    // expect(wrapper.find('h1').text()).toBe('Expensify');
    
    // const renderer = new ShallowRenderer;
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
    // console.log(renderer.getRenderOutput());
})