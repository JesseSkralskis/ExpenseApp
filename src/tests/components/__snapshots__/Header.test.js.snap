// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header corectly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
    create
  </NavLink>
</header>
`;
